app/jobs/bigquery_report_export_job.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "google/cloud/bigquery"
+
+class BigqueryReportExportJob < ApplicationJob
+  DATASET_NAME = "daily"
+  ACCOUNTS_TABLE_NAME = "accounts"
+  EVENTS_TABLE_NAME = "events"
+
+  queue_as :default
+
+  def perform(start_date, end_date)
+    bigquery = Google::Cloud::Bigquery.new(credentials: Rails.application.secrets.bigquery_credentials)
+    dataset = bigquery.dataset DATASET_NAME
+
+    replace_accounts_table(dataset: dataset)
+    update_events_table(dataset: dataset, start_date: start_date, end_date: end_date)
+  end
+
+protected
+
+  def replace_accounts_table(dataset:)
+    report = Report::Accounts.report(
+      user_id_pepper: Rails.application.secrets.reporting_user_id_pepper,
+    )
+
+    delete_job = dataset.query_job "DELETE * FROM #{ACCOUNTS_TABLE_NAME}"
```

Comment:
Pretty cool that you can just submit SQL-like syntax like that.

How does this work?

Are there limits to what you can fire through the API like that? Like your credentials are enabled for delete and/or read and/or write?

Reply:
Yeah, the credentials will be limited in what they can access.  It's all using role-based access control on the GCP side.

Reply:
I've made a service account which I think has permissions to just delete and load the data.  But apparently I don't have permissions to see (or edit) its permissions now that it's been made!  Well, we can cross that bridge when we come to it...